File: server/src/main/java/com/example/schoolsite/entity/Calendar.java
package com.example.schoolsite.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "calendar")
@ToString
public class Calendar {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Column(name = "semesterID", nullable = false)
    private int semesterID;
    @Column(name = "weekDay", nullable = false)
    private int weekDay;
    @Column(name = "lessonNumber", nullable = false)
    private int lessonNumber;

    public Calendar(int semesterID, int weekDay, int lessonNumber) {
        this.semesterID = semesterID;
        this.weekDay = weekDay;
        this.lessonNumber = lessonNumber;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Calendar calendar = (Calendar) o;
        return weekDay == calendar.weekDay && lessonNumber == calendar.lessonNumber && Objects.equals(id, calendar.id) && Objects.equals(semesterID, calendar.semesterID);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, semesterID, weekDay, lessonNumber);
    }
}

File: server/src/main/java/com/example/schoolsite/entity/Calendar.kt
package com.example.schoolsite.entity

import lombok.Getter

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "calendar")
@ToString
class Calendar(@field:Column(name = "semesterID", nullable = false) private val semesterID: Int, @field:Column(name = "week_day", nullable = false) private val weekDay: Int, @field:Column(name = "lesson_number", nullable = false) private val lessonNumber: Int) {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private val id: Long? = null

    @OneToOne(optional = false, mappedBy = "calendarId")
    private val shedule: Shedule? = null

    @Override
    override fun equals(o: Object?): Boolean {
        if (this === o) return true
        if (o == null || getClass() !== o.getClass()) return false
        val calendar = o as Calendar
        return weekDay == calendar.weekDay && lessonNumber == calendar.lessonNumber && Objects.equals(id, calendar.id) && Objects.equals(semesterID, calendar.semesterID)
    }

    @Override
    override fun hashCode(): Int {
        return Objects.hash(id, semesterID, weekDay, lessonNumber)
    }
}